mlflow/models/evaluation/base.py

```diff
@@ -643,6 +659,101 @@ def _get_last_failed_evaluator():
     return _last_failed_evaluator
 
 
+def _validate(validation_thresholds, candidate_metrics, baseline_metrics=None):
+    """
+    Validate the model based on validation_thresholds by metrics value and
+    metrics comparison between candidate model's metrics (candidate_metrics) and
+    baseline_model's metrics (baseline_metrics).
+    :param validation_thresholds: A dictionary from metric_name to MetricThreshold.
+    :param candidate_metrics: The metric evaluation result of the candidate model.
+    :param baseline_metrics: The metric evaluation result of the baseline model.
+    If the validation does not pass, raise an MlflowException with detail failure message.
+    """
+    if not baseline_metrics:
+        baseline_metrics = dict()
+
+    validation_results = {
+        metric_name: _MetricValidationResult(
+            metric_name,
+            candidate_metrics.get(metric_name, None),
+            threshold,
+            baseline_metrics.get(metric_name, None),
+        )
+        for (metric_name, threshold) in validation_thresholds.items()
+    }
+
+    for metric_name in validation_thresholds.keys():
+
+        metric_threshold, validation_result = (
+            validation_thresholds[metric_name],
+            validation_results[metric_name],
+        )
+
+        if metric_name not in candidate_metrics:
+            validation_result.missing_candidate = True
+            continue
+
+        candidate_metric_value, baseline_metric_value = (
+            candidate_metrics[metric_name],
+            baseline_metrics[metric_name] if baseline_metrics else None,
+        )
+
+        # If metric is higher is better, >= is used, otherwise <= is used
+        # for thresholding metric value and model comparsion
+        comparator_fn = operator.__ge__ if metric_threshold.higher_is_better else operator.__le__
+
+        if metric_threshold.threshold is not None:
+            # metric threshold failed
+            # - if not (metric_value >= threshold) for higher is better
+            # - if not (metric_value <= threshold) for lower is better
+            validation_result.threshold_failed = not comparator_fn(
+                candidate_metric_value, metric_threshold.threshold
+            )
+
+        if (
+            metric_threshold.min_relative_change or metric_threshold.min_absolute_change
+        ) and metric_name not in baseline_metrics:
+            validation_result.missing_baseline = True
+            continue
+
+        if metric_threshold.min_absolute_change is not None:
+            # metric comparsion aboslute change failed
+            # - if not (metric_value >= baseline + min_absolute_change) for higher is better
+            # - if not (metric_value <= baseline + min_absolute_change) for lower is better
```

Comment:
```suggestion
            # - if not (metric_value <= baseline - min_absolute_change) for lower is better
```

Reply:
There seems to be an asymmetry here; for "higher is better", it appears that we don't allow a metric to pass validation unless it's `min_absolute_change` better than `baseline`. However, for "lower is better", we allow metrics to pass validation as long as they're no more than `min_absolute_change` worse than `baseline`.

Reply:
Can we address this and add test coverage?

Reply:
Actually this is because we intend to let users to define min_absolute_change to be positive for higher is better and negative for lower is better; we can discuss whether we want that or we want min_absolute_change to be always negative?

Reply:
Do you mean always be positive? That would make more sense to me.

Reply:
Made `min_absolute_change` to be always positive